packages/core/src/ruleset/validation/ajv.ts

```diff
@@ -0,0 +1,66 @@
+import Ajv, { _, ValidateFunction } from 'ajv';
```

Comment:
Largely copied from `packages/core/src/ruleset/validation.ts` with a few subtle adjustments to account for 2 different rulesets formats.